pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -259,6 +272,25 @@ func (s *Service) Disconnected(_ network.Network, c network.Conn) {
 	delete(s.receivedHandshakes, c.RemotePeer())
 }
 
+// SetWelcomeMessage sets the new handshake welcome message.
+func (s *Service) SetWelcomeMessage(msg string) (err error) {
+	if len(msg) > MaxWelcomeMessageLength {
+		return ErrWelcomeMessageLength
+	}
+	s.welcomeMessage.mu.Lock()
+	defer s.welcomeMessage.mu.Unlock()
+	s.welcomeMessage.val = msg
+	return nil
+}
+
+// WelcomeMessageSynced returns the synced value of the current welcome message.
+func (s *Service) WelcomeMessageSynced() string {
```

Comment:
this is a classic getter. no need to use `Synced` in the future. `GetWelcomeMessage` would suffice in the future as a naming convention. please note my comment about `atomic.Value` for this purpose as it would suffice and make the type redundant